pkg/release/release.go

```diff
@@ -364,18 +364,26 @@ next:
 	case TestAction:
 		if hr.Spec.Test.Enable {
 			logger.Log("info", "running test", "action", TestAction)
+			status.SetStatusPhase(r.hrClient.HelmReleases(hr.Namespace), hr, apiV1.HelmReleasePhaseTesting)
 			if err = r.test(client, hr); err != nil {
 				logger.Log("error", err, "action", TestAction)
 				errs = append(errs, err)
-
-				if curRel == nil {
-					action = UninstallAction
+				status.SetStatusPhase(r.hrClient.HelmReleases(hr.Namespace), hr, apiV1.HelmReleasePhaseTestFailed)
+
+				if !hr.Spec.Test.GetIgnoreFailures() {
+					if curRel == nil {
+						action = UninstallAction
+					} else {
+						action = RollbackAction
+					}
 				} else {
-					action = RollbackAction
+					action = AnnotateAction
+					logger.Log("info", "test failed - ignoring failures", "ignoreFailures", hr.Spec.Test.GetIgnoreFailures(), "action", action)
 				}
 				goto next
 			}
 
+			status.SetStatusPhase(r.hrClient.HelmReleases(hr.Namespace), hr, apiV1.HelmReleasePhaseTested)
 			logger.Log("info", "test succeeded", "revision", chart.revision, "action", action)
 		}
 
```

Comment:
The line directly below this (L390) should be called in the case of ignored test failures, but AFAICT is not due to L383, which I believe needs to be moved to directly after L378, and then I think L386-L387 need to put in an `else` block (so it's only called when tests succeed).

Reply:
agreed